hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/AlertsEngineImpl.java

```diff
@@ -762,19 +773,21 @@ private void handleAutoResolvedTriggers() {
 
     private int checkMissingStates() {
         int numMatchingEvals = 0;
-        for (MissingState missingState : missingStates) {
+        Iterator<MissingState> itMissingStates = missingStates.iterator();
```

Comment:
I will believe you.  Although, I always thought the for-each construct was just syntactic sugar on top of an underlying Iterator, which would make this change a no-op.  Whatever is happening, it seems subtle in the code and may merit an inline comment about the concurrency issue and how this is safe.